buildkite.sh

```diff
@@ -0,0 +1,356 @@
+#!/bin/bash
+
+cat << 'EOF'
+steps:
+  - command: |
+      uname -a
+      make --version
+      \${SHELL} --version || true
+      c++ --version
+      ld -v
+      ! command -v gdb &>/dev/null || gdb --version
+      dmd --version || true
+    label: "Print envs"
+    env:
+      DETERMINISTIC_HINT: 1
+
+  - command: |
+      set -uexo pipefail
+      origin_repo=$(echo "$BUILDKITE_REPO" | sed "s/.*\/\([^\]*\)[.]git/\1/")
+      origin_target_branch="\$BUILDKITE_PULL_REQUEST_BASE_BRANCH"
+      if [ -z "\$origin_target_branch" ] ; then
+        origin_target_branch="\$BUILDKITE_BRANCH"
+      fi
+      echo \$origin_target_branch
+
+      for dir in dmd druntime phobos tools dub ; do
+          rm -rf \$dir
+          if [ "\$origin_repo" == "\$dir" ] ; then
+            # we have already cloned this repo, so let's use this data
+            mkdir -p \$dir
+            cp -r \$(ls -A | grep -v dmd) \$dir
+          else
+            branch=\$(git ls-remote --exit-code --heads https://github.com/dlang/\$dir "\${origin_target_branch}" > /dev/null || echo "master")
+            git clone -b "\${branch:-master}" --depth 1 https://github.com/dlang/\$dir
+          fi
+      done
+
+      for dir in dmd druntime phobos ; do
+          make -C \$dir -f posix.mak AUTO_BOOTSTRAP=1 --jobs=4
+      done
+
+      # build dub
+      # TODO: doesn't build with master
+      (cd dub;
+      . $(curl i.dlang.io | bash -s dmd-2.079.1 -a)
```

Comment:
Is it worth relying on your private server to abbreviate this?
Also `i.dlang.io` does not forward to https.

Reply:
No this was just me abbreviating thing. Sorry. Changed to https://dlang.org/install.sh
Though of course ideally we figure out why it doesn't build at the moment.